lib/bedrock-item.js

```diff
@@ -0,0 +1,207 @@
+          has_nbt: this.nbt !== null,
+          nbt: this.nbt !== null ? { version: 1, nbt: this.nbt } : undefined,
+          can_place_on: this.canPlaceOn,
+          can_destroy: this.canDestroy
+        }
+      }
+    }
+
+    /** @returns {BedrockItem} */
+    static fromNetwork (item, serverAuthoritative = true) {
+      return new BedrockItem(
+        item.network_id,
+        item.count,
+        item.metadata,
+        item.extra.nbt?.nbt ?? null,
+        item.extra.can_place_on,
+        item.extra.can_destroy,
+        item.stack_id
+      )
+    }
+
+    get customName () {
+      if (Object.keys(this).length === 0) return null
+      return this.nbt?.value?.display?.value?.Name?.value ?? null
+    }
+
+    set customName (newName) {
+      if (!this.nbt) this.nbt = { name: '', type: 'compound', value: {} }
+      if (!this.nbt.value.display) this.nbt.value.display = { type: 'compound', value: {} }
+      this.nbt.value.display.value.Name = { type: 'string', value: newName }
+    }
+
+    get customLore () {
+      if (Object.keys(this).length === 0) return null
+      return this.nbt?.value?.display?.value?.Lore?.value.value ?? null
+    }
+
+    set customLore (newLore) {
+      if (!this.nbt) this.nbt = { name: '', type: 'compound', value: {} }
+      if (!this.nbt.value.display) this.nbt.value.display = { type: 'compound', value: {} }
+      this.nbt.value.display.value.Lore = { type: 'list', value: { type: 'string', value: newLore } }
+    }
+
+    get repairCost () {
+      if (Object.keys(this).length === 0) return 0
+      return this.nbt?.value.RepairCost?.value ?? 0
+    }
+
+    set repairCost (value) {
+      if (!this.nbt) this.nbt = { name: '', type: 'compound', value: {} }
+      this.nbt.value.RepairCost = { type: 'int', value }
+    }
+
+    get enchants () {
+      if (Object.keys(this).length === 0) return 0
+      if (!this.nbt?.value?.ench) return []
+      return this.nbt.value.ench.value.value.map((ench) => ({
+        lvl: ench.lvl.value,
+        name: registry.enchantments[ench.id.value]?.name || null // TODO: bedrock enchantments
+      }))
+    }
+
+    set enchants (normalizedEnchArray) {
+      const enchs = normalizedEnchArray.map(({ name, lvl }) => ({
+        id: { type: 'short', value: registry.enchantmentsByName[name].id },
+        lvl: { type: 'short', value: lvl }
+      }))
+
+      if (enchs.length !== 0) {
+        if (!this.nbt) this.nbt = { name: '', type: 'compound', value: {} }
+        this.nbt.value.ench = { type: 'list', value: { type: 'compound', value: enchs } }
+      }
+    }
+
+    get durabilityUsed () {
+      if (Object.keys(this).length === 0) return null
+      return this.nbt?.value?.Damage?.value ?? 0
+    }
+
+    set durabilityUsed (value) {
+      if (!this.nbt) this.nbt = { name: '', type: 'compound', value: {} }
+      this.nbt.value.Damage = { type: 'int', value }
+    }
+
+    get spawnEggMobName () {
+      return this.name.replace('_spawn_egg', '')
```

Comment:
is this accurate?

Reply:
As far as I have tested on the current version. Probably not on older versions, but I would need to test more for that (spawneggs is mostly why I added it to the todo list)

Reply:
The mob name is in the spawn egg item's name from 1.16.100, according to [the wiki](https://minecraft.fandom.com/wiki/Spawn_Egg#History). Probably could add this as the `spawnEggsHaveSpawnedEntityInName` feature in minecraft-data